R/plotAbundanceDensity.R

```diff
@@ -0,0 +1,259 @@
+                             xlab = NULL,
+                             ylab = NULL,
+                             colour_by = NULL,
+                             point_alpha = 0.6,
+                             point_shape = 21,
+                             point_size = 2,
+                             add_legend = TRUE,
+                             flipped = FALSE,
+                             add_x_text = TRUE,
+                             log_scale_x_axis = FALSE){
+    # start plotting
+    # Density plot needs different kind of structure
+    if (layout == "density"){
+        # # Reorders the levels to reverse order so that the taxa with highest abundance
+        # is on the top
+        density_data$Y <- factor(density_data$Y , levels = rev(levels(density_data$Y)) )
+        plot_out <- ggplot(density_data, aes_string(x="X", colour = "Y")) +
+            xlab(xlab) +
+            ylab(ylab)
+    }
+    else {
+        plot_out <- ggplot(density_data, aes_string(x="X", y="Y")) +
+            xlab(xlab) +
+            ylab(ylab)
+    }
+    
+    # Layout can be "density", "jitter", or "point"
+    if (layout == "point"){
+        density_out <- .get_point_args(colour_by,
+                                       shape_by = NULL,
+                                       size_by = NULL,
+                                       alpha = point_alpha,
+                                       shape = point_shape,
+                                       size = point_size,
+                                       position = "identity")
+        plot_out <- plot_out +
+            do.call(geom_point, density_out$args)
+    }
+    else if (layout == "density"){
+        density_out <- .get_density_args(alpha = 0.65)
+        plot_out <- plot_out +
+            do.call(geom_density, density_out$args) + 
+            facet_grid(Y~., switch = "y", scales="free") + 
+            theme_classic() +
+            theme(strip.background = element_blank(), strip.text.y.left = element_text(angle = 0), # Removes label grid, horizontal labels
+                  axis.ticks.y = element_blank(), axis.text.y = element_blank(), # Removes y-axis
+                  axis.title.y = element_blank(), axis.line.y = element_blank()) # Removes y-axis
+        # colour_by and flipped are disabled in density_plot
+        colour_by <- NULL
+        flipped <- FALSE
+    }
+    else {
+        density_out <- .get_point_args(colour_by,
+                                       shape_by = NULL,
+                                       size_by = NULL,
+                                       alpha = point_alpha,
+                                       shape = point_shape,
+                                       size = point_size,
+                                       position = "jitter")
+        plot_out <- plot_out +
+            do.call(geom_point, density_out$args)
+    }
+    if (log_scale_x_axis){
+        plot_out <- plot_out + scale_x_log10() 
```

Comment:
a log scaled x axis should not be added in here. The user can always add it outside, if he needs to by actually just adding one element to the plot. I think this is not worth running the argument through all the functions.

Reply:
also good

Reply:
Removed